crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -556,32 +607,56 @@ fn find_attrs(mut expr: Expr) -> Result<AttrSet, ()> {
 }
 
 /// Copy a whole directory recursively ignoring the original permissions
-async fn copy_dir_recursively_without_permissions(
+///
+/// We need this because:
+/// 1. Sometimes we need to copy from the Nix store
+/// 2. fs_extra::dir::copy doesn't handle symlinks.
+///    See: https://github.com/webdesus/fs_extra/issues/61
+async fn copy_dir(
     from: &impl AsRef<Path>,
     to: &impl AsRef<Path>,
+    keep_permissions: bool,
 ) -> Result<(), std::io::Error> {
     for entry in WalkDir::new(from).into_iter().skip(1) {
         let entry = entry.unwrap();
         let new_path = to.as_ref().join(entry.path().strip_prefix(from).unwrap());
-        if entry.file_type().is_dir() {
-            tokio::fs::create_dir(new_path).await.unwrap()
-        } else {
-            copy_file_without_permissions(entry.path(), &new_path)
-                .await
-                .unwrap()
+        match entry.file_type() {
+            file_type if file_type.is_dir() => {
+                tokio::fs::create_dir(new_path).await.unwrap();
```

Comment:
Let's not use `async` unless it's absolutely necessary. Is there a reason `std::fs::create_dir` doesn't satisfy your needs?

Reply:
Dropped a bunch of async stuff